packages/template-retail-react-app/app/components/product-view/index.jsx

```diff
@@ -215,12 +239,21 @@ const ProductView = forwardRef(
                 addToWishlist(product, variant, quantity)
             }
 
-            if (addToCart || updateCart) {
+            let disableButton = showInventoryMessage
+            if ((isProductASet || isProductABundle) && childProductOrderability) {
+                // if any of the children are not orderable, it will disable the add to cart button
+                Object.keys(childProductOrderability).forEach((productId) => {
+                    disableButton = !childProductOrderability[productId] || disableButton
+                })
```

Comment:
We can simply this logic with Array's [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).
```suggestion
                disableButton = Object.keys(childProductOrderability).some(
                    (productId) => !childProductOrderability[productId]
                )
```